MiguelAValdesG/SpringDataJPA/src/main/java/com/bootcamp/springdatajpa/service/impl/AddressServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.bootcamp.springdatajpa.service.impl;
+
+import com.bootcamp.springdatajpa.dao.AddressDAO;
+import com.bootcamp.springdatajpa.dao.OwnerDAO;
+import com.bootcamp.springdatajpa.dto.AddressDTO;
+import com.bootcamp.springdatajpa.entity.Address;
+import com.bootcamp.springdatajpa.entity.Owner;
+import com.bootcamp.springdatajpa.exception.InvalidDataException;
+import com.bootcamp.springdatajpa.mapper.AddressMapper;
+import com.bootcamp.springdatajpa.service.AddressService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.bootcamp.springdatajpa.utils.Constants.ADDRESS_NOT_FOUND;
+import static com.bootcamp.springdatajpa.utils.Constants.ADDRESS_CANNOT_BE_DELETED;
+
+@Service
+public class AddressServiceImpl implements AddressService {
+
+  @Autowired
+  AddressDAO addressDAO;
+  @Autowired
+  OwnerDAO ownerDAO;
+
+  @Autowired
+  AddressMapper addressMapper;
+
+  @Override
+  public List<AddressDTO> getAllAddresses() {
+    return addressDAO.findAll().parallelStream()
```

Comment:
why parallelStream() ?

Reply:
I don't know why not, I use the stream for do a "traverse" through the response of findAll of my DAO for "convert" all the entities in DTO through the entityToDTO mapper method.
The parallel is only for to do it in different threads. 

Should I do only with stream? Or I should do it of different manner?

Reply:
https://www.baeldung.com/java-when-to-use-parallel-stream